container.go

```diff
@@ -50,6 +53,20 @@ type Container struct {
 	graph.Graph
 }
 
+// InvokeOnce only allows function invokation once to register the
+// return types. If return types are already registered, specific error
+// is returned to the caller
+func (c *Container) InvokeOnce(t interface{}) error {
```

Comment:
What you mean its not going to work? The test you mentioned will fail as It expects `ErrInvokeOnce` on second invoke. I'll add your test case.